File: examples/payload_set_stream_bitrate.cpp
/**
 * This sample will send the command to change the config for the bitrate of the video stream
 * The change will be applied immediately
 **/

#include "stdio.h"
#include"payloadSdkInterface.h"

#include <iostream>
#include <chrono>

PayloadSdkInterface* my_payload = nullptr;

#if (CONTROL_METHOD == CONTROL_UART)
T_ConnInfo s_conn = {
    CONTROL_UART,
    payload_uart_port,
    payload_uart_baud
};
#else
T_ConnInfo s_conn = {
    CONTROL_UDP,
    udp_ip_target,
    udp_port_target
};
#endif

void quit_handler(int sig);


int main(int argc, char *argv[]){
	printf("Starting SendSystemTime example...\n");
	signal(SIGINT,quit_handler);

	// creat payloadsdk object
	my_payload = new PayloadSdkInterface(s_conn);

	// init payload
	my_payload->sdkInitConnection();
	printf("Waiting for payload signal! \n");

	// check connection
	my_payload->checkPayloadConnection();

	int msg_cnt = 0;
	int boot_time_ms = 0;

	while(1){
		
		uint32_t bitrate = 8000000; // bit per second
		my_payload->setPayloadStreamBitrate(bitrate);

		printf("The stream bitrate was set to %d. Exit \n", bitrate);

		usleep(1000000); // 1000ms, 1Hz
		exit(0);
	}

	return 0;
}

void quit_handler( int sig ){
    printf("\n");
    printf("TERMINATING AT USER REQUEST \n");
    printf("\n");

    // close payload interface
    try {
        my_payload->sdkQuit();
    }
    catch (int error){}

    // end program here
    exit(0);
}

File: examples/payload_set_stream_bitrate.py
#!/usr/bin/env python3
import sys
import os

# Add the libs directory to the path
sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'libs'))

# Import config first to setup environment automatically
from config import config

import time
import signal
from payload_sdk import PayloadSdkInterface, mavlink_system_time_t
from payload_define import *

my_payload = None

# Signal handler for quitting
def quit_handler(sig, frame):
    global my_payload
    print("\nTERMINATING AT USER REQUEST")

    # Close payload interface
    try:
        my_payload.sdkQuit()
    except Exception as e:
        print(f"Error while quitting payload: {e}")

    # End program    
    sys.exit(0)

def main():
    global my_payload

    print("Starting SendSystemTime example...")
    signal.signal(signal.SIGINT, quit_handler)

    # Create payloadsdk object
    my_payload = PayloadSdkInterface()

    # Init payload
    my_payload.sdkInitConnection()
    print("Waiting for payload signal!")

    # Check connection
    my_payload.checkPayloadConnection()

    while True:
        bitrate = 1000000 # bit per second
        my_payload.setPayloadStreamBitrate(bitrate)
        print(f"The stream bitrate was set to", bitrate)
        time.sleep(1)  # 1000ms ~ 1Hz

        sys.exit(0)

if __name__ == "__main__":
    main()